react-time-track/src/components/add-member-project.js

```diff
@@ -0,0 +1,302 @@
+/**@jsx jsx */
+import React from "react";
+import { jsx } from "@emotion/core";
+import { FaAngleDown, FaAngleUp } from "react-icons/fa";
+
+import { Button } from "./ui";
+
+function AddMemberProject({ listMember, addMemberFn, closeModalFn }) {
+  let infoProject = JSON.parse(sessionStorage.getItem("InfoNewProject"));
+  const startDate = new Date(infoProject.start.split("-").join("/"));
+  const endDate = new Date(infoProject.end.split("-").join("/"));
+
+  const [selectedMember, setSelectedMember] = React.useState(listMember[0]);
+  const [availableTime, setAvailableTime] = React.useState(0);
+
+  const millisecondsInADay = 86400000;
+  const formatDate = { day: "2-digit", month: "2-digit", year: "2-digit" };
+  const getTimeWeekly = React.useMemo(() => {
+    let startNumberDay = startDate.getDay();
+    return selectedMember.availableTime.reduce(
+      (result, value) => {
+        let lastElement = result.slice(-1)[0];
+        if (startNumberDay > 5) {
+          startNumberDay = 1;
+          let newStartDate = new Date(
+            lastElement[1].getTime() + 3 * millisecondsInADay
+          );
+          result.push([newStartDate, newStartDate, value]);
+        } else {
+          if (value < lastElement[2]) lastElement[2] = value;
+
+          lastElement[1] = new Date(
+            lastElement[1].getTime() + millisecondsInADay
+          );
+        }
+        startNumberDay++;
+        return result;
+      },
+      [
+        [
+          startDate,
+          new Date(startDate.getTime() - millisecondsInADay),
+          selectedMember.availableTime[0]
+        ]
+      ]
+    );
+  }, [selectedMember.availableTime, startDate]);
+
+  const [detailTimeByWeek, setDetailTimeByWeek] = React.useState(getTimeWeekly);
+  const [isShownDetailTime, setIsShownDetailTime] = React.useState(false);
+
+  function handleChangeMember(event) {
+    setSelectedMember(listMember[event.target.selectedIndex]);
+  }
+
+  function handleCancelMember() {
+    closeModalFn();
+  }
+
+  function handleExpandCollapse() {
+    setIsShownDetailTime(!isShownDetailTime);
+  }
+
+  function handleSubmitMember(event) {
+    event.preventDefault();
+
+    const ddlEmployee = event.target.elements.ddlEmployee;
+    let id = +ddlEmployee.value;
```

Comment:
Personally I prefer to use parseInt or parseFloat but this is shorter and practical.